src/Polysemy/Error.hs

```diff
@@ -242,14 +244,24 @@ mapError f = interpretH $ \case
 {-# INLINE mapError #-}
 
 
-newtype WrappedExc e = WrappedExc { unwrapExc :: e }
-  deriving (Typeable)
+data WrappedExc = WrappedExc !Unique Any
```

Comment:
Stick an `{-# UNPACK #-}` on the `Unique`. (Unless it's a no-op? I don't think it is for `Unique`s.)

Reply:
It is a no-op, since `Unique` is already a `newtype` by itself.